vault/testing.go

```diff
@@ -1633,6 +1635,7 @@ func NewTestCluster(t testing.T, base *CoreConfig, opts *TestClusterOptions) *Te
 	}
 
 	if base != nil {
+		coreConfig.DetectDeadlocks = TestDeadlockDetection
```

Comment:
This is very close to what I want.  The only catch is, there's no way to turn it off.  Maybe that's ok though, I can't come up with a good reason why I should have to be able to, given that I can use the tag.  We can always revisit this later if need be.